src/integration-test/java/org/rulez/demokracia/pdengine/integrationtest/VoteCreationIntegrationTest.java

```diff
@@ -1,140 +1,133 @@
+)
+public class VoteCreationIntegrationTest {
+
+  private static final String ASSURANCE_NAME = "assurance";
+  @Autowired
+  private MockMvc mvc;
+
+  @Autowired
+  private VoteService voteService;
+  private CreateVoteRequest voteRequest;
+  private CreateVoteRequest badVoteRequest;
 
-  @Override
   @Before
   public void setUp() {
-    thread = new JettyThread();
-    thread.run();
-    initializeCreateVoteRequest();
-    super.setUp();
+    voteRequest = initializeCreateVoteRequest("voteName");
+    badVoteRequest =
+        initializeCreateVoteRequest("`drop table little_bobby tables;`");
   }
 
   @Test
-  public void vote_can_be_created_through_rest_interface() {
-    final Invocation.Builder invocationBuilder = createWebClient();
-    final Response response =
-        invocationBuilder.post(Entity.entity(req, MediaType.APPLICATION_JSON));
-    final VoteAdminInfo adminInfo = response.readEntity(VoteAdminInfo.class);
-    final VoteEntity vote = voteManager.getVote(adminInfo.voteId);
-    assertVoteNameAndKeyIsCorrect(adminInfo, vote);
-  }
+  public void vote_can_be_created_through_rest_interface() throws Exception {
+    MvcResult result = callEndpointWithRequest(voteRequest)
+        .andExpect(status().isOk())
+        .andReturn();
 
-  private void assertVoteNameAndKeyIsCorrect(
-      final VoteAdminInfo adminInfo, final VoteEntity vote
-  ) {
-    assertEquals(req.voteName, vote.name);
-    assertEquals(adminInfo.adminKey, vote.adminKey);
-  }
+    VoteAdminInfo info = new Gson().fromJson(
+        result.getResponse().getContentAsString(), VoteAdminInfo.class
+    );
 
-  @Test
-  public void vote_creation_fails_with_404_on_bad_input() {
-    final Response response = restCallWithBadInput();
-    assertEquals(400, response.getStatus());
-  }
+    Vote savedVote = voteService.getVoteWithValidatedAdminKey(info);
 
-  @Test
-  public void vote_creation_fails_and_reports_error_message_on_bad_input() {
-    final Response response = restCallWithBadInput();
-    final String responseString = response.readEntity(String.class);
-    final JsonObject responseJson =
-        new JsonParser().parse(responseString).getAsJsonObject();
-    assertEquals(
-        "invalid characters in {0}", responseJson
-            .get("error").getAsJsonObject()
-            .get("message").getAsString()
-    );
+    assertEquals(voteRequest.getVoteName(), savedVote.getName());
   }
 
   @Test
-  public void vote_creation_fails_and_reports_error_details_on_bad_input() {
-    final Response response = restCallWithBadInput();
-    final String responseString = response.readEntity(String.class);
-    final JsonObject responseJson =
-        new JsonParser().parse(responseString).getAsJsonObject();
-    assertEquals(
-        "vote name", responseJson
-            .get("error").getAsJsonObject()
-            .get("details").getAsString()
-    );
+  @SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
```

Comment:
get rid of this and the following supressWarnings.
disable JUnitTestsShouldIncludeAssert in pmd_rules.xml,
and add this rule:
```
<rule name="JUnitTestsShouldIncludeAssert"
      language="java"
      message="Junit tests should have checks"
      class="net.sourceforge.pmd.lang.rule.XPathRule"
      <!-- externalInfoUrl="TODO"--> >
    <description>

    </description>
    <priority>3</priority>
    <properties>
        <property name="xpath">
            <value>
<![CDATA[
//MethodDeclaration[
  ./preceding-sibling::Annotation[.//@Image='Test'] and
  not(
    .//Name[matches(@Image,'assert')] or
    .//PrimarySuffix[matches(@Image,'andExpect')]
  )
]

]]>
            </value>
        </property>
    </properties>
    <!--<example><![CDATA[]]></example>-->
</rule>
```